content/java/concepts/hashmap/hashmap.md

```diff
@@ -6,28 +6,26 @@ Subjects:
 Tags:
   - 'HashMap'
   - 'Data Types'
+  - 'Collections'
+  - 'Data Structures'
 CatalogContent:
   - 'learn-java'
   - 'paths/computer-science'
 ---
 
-In Java, a `HashMap` is used to store items as key-value pairs. The key and value can be of the same or different data type.
+A `HashMap`, like a [`TreeMap`](https://www.codecademy.com/resources/docs/java/TreeMap), implements the [`Map`](https://www.codecademy.com/resources/docs/java/Map) interface, which means it stores items as key-value pairs. Unlike a `TreeMap`, the `HashMap` stores its items sorted in no particular order. (Actually, it's sorted by the hashes of its keys, which for most purposes is essentially random.) This is due to the different storage types each collection uses internally. The `HashMap` stores its keys as hashes for lookup, whereas the `TreeMap` stores its keys in a binary tree structure for lookup. The advantage of a `HashMap` is that its operations are much faster.
 
 ## Syntax
 
 The `HashMap` class comes from the `java.util` package. Therefore, it must be imported in order to be used:
 
 ```java
 import java.util.HashMap
-```
-
-Next, the `HashMap` is initialized with two generic types inside angle brackets `<` `>`:
 
-```java
 HashMap<KeyDatatype, ValueDatatype> myHashMap = new HashMap<KeyDatatype, ValueDatatype>();
 ```
 
-The generic data type for `KeyDatatype` and `ValueDatatype` can either be different or the same.
+The `HashMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDatatype` and `ValueDatatype` can either be different or the same.
 
 ## Keys and Values
 
```

Comment:
Below this line, I think that since the .get() method is first introduced here, maybe we could refer to it by name here and omit the entire "Accessing an Item" section since that's what this snippet seems to be doing?

Reply:
Also, I wonder if the comment on line # 43 is really necessary? There doesn't seem to be a variable called `CourseNames` or `Teacher` anywhere around?